frontend/src/pages/modelRegistry/screens/RegisterModel/RegisterModel.tsx

```diff
@@ -36,12 +37,21 @@ const RegisterModel: React.FC = () => {
   const [isSubmitting, setIsSubmitting] = React.useState(false);
   const [submitError, setSubmitError] = React.useState<Error | undefined>(undefined);
   const [formData, setData] = useRegisterModelData();
-  const isModelNameValid = isNameValid(formData.modelName);
-  const isSubmitDisabled = isSubmitting || isRegisterModelSubmitDisabled(formData);
-  const { modelName, modelDescription } = formData;
   const [registeredModelName, setRegisteredModelName] = React.useState<string>('');
   const [versionName, setVersionName] = React.useState<string>('');
   const [errorName, setErrorName] = React.useState<string | undefined>(undefined);
+  const [registeredModels] = useRegisteredModels();
+
+  const doesModelNameExist = (name: string) =>
+    registeredModels.items.some((model) => model.name === name);
+
+  const isModelNameValid =
+    isNameValid(formData.modelName) && !doesModelNameExist(formData.modelName);
+  const isSubmitDisabled =
+    isSubmitting ||
+    isRegisterModelSubmitDisabled(formData) ||
```

Comment:
Hmm. We're slightly violating the abstraction of `isRegisterModelSubmitDisabled` here, ideally it should encapsulate the `doesModelNameExist` logic too. But we would have to either pass the `registeredModels` array into it or pass the `doesModelNameExist` callback into it.

Can we do the former? Maybe move `doesModelNameExist` to `frontend/src/pages/modelRegistry/screens/RegisterModel/utils.ts`, have it take the `registeredModels` as another arg, and call it from inside `isRegisterModelSubmitDisabled` and in the other places where we need it here.

I'm also wondering if we need to do this same validation for the version name... but maybe we can save that for a separate bug if it comes up.

Reply:
We could replace the `isNameValid` function with `validateName` and have it return an error message or undefined if no error, then we can just use its truthiness for whether to render the error and its value for the error?

But we don't necessarily have to do that to put `doesModelNameExist` logic inside `isRegisterModelSubmitDisabled`, we already call `isNameValid` both inside and outside of that function. I may be nitpicking too much based on the name of the function, it is only called in one place - I think the intention was just to make the form-level validation unit testable which we didn't do (yet)